Updating markdown files (#24)
microsoft/python-environment-tools: README.md

# Python environment tools for Visual Studio Code

> Performant Python environment tooling and support, such as locating all global Python installs and virtual environments. Powers environment environment tooling for the [Python extension](https://marketplace.visualstudio.com/items?itemName=ms-python.python) in [Visual Studio Code](https://code.visualstudio.com/).

Author: Perhaps we should exclude the second sentence since for now since that is not the current reality.

Reviewer: I agree

## Environment Types Supported 

> - Windows Registry
> - Windows Store
> - PyEnv
> - Conda
> - PipEnv
> - Homebrew
> - VirtualEnvWrapper
> - Venv
> - VirtualEnv

Reviewer: ```suggestion
- python.org
- Windows Store
- PyEnv
- PyEnv-Win
- PyEnv-Virtualenv
- Conda
- PipEnv
- Homebrew
- VirtualEnvWrapper
- VirtualEnvWrapper-Win
- Venv
- VirtualEnv
- Python on your PATH
```

# Python environment tools for Visual Studio Code

Performant Python environment tooling and support, such as locating all global Python installs and virtual environments.

## Environment Types Supported 

- Windows Registry
- Windows Store
- PyEnv
- Conda
- PipEnv
- Homebrew
- VirtualEnvWrapper
- Venv
- VirtualEnv

## Features 

- Discovery of all global Python installs
- Discovery of all Python virtual environments

## Key Methodology

> We leverage Rust's performance to quickly scan for Python environments available to the user. When we find them, we gather all relevant information immediately to avoid the need for repeated discovery.

Author: Would like to call out out methodology and "principles" used in this work. Will likely need some additional details to ensure this is comprehensive

# Python environment tools for Visual Studio Code

Performant Python environment tooling and support, such as locating all global Python installs and virtual environments.

## Environment Types Supported 

- python.org
- Windows Store
- PyEnv
- PyEnv-Win
- PyEnv-Virtualenv
- Conda
- Miniconda
- Miniforge
- PipEnv
- Homebrew
- VirtualEnvWrapper
- VirtualEnvWrapper-Win
- Venv
- VirtualEnv
- Python on your PATH

## Features 

- Discovery of all global Python installs
- Discovery of all Python virtual environments

## Key Methodology

> We leverage Rust's performance to quickly scan for Python environments available to the user. When we find them, we gather all relevant information immediately to avoid the need for repeated discovery.

Author: ```suggestion
Our approach is focused on performance and efficiency which leverages Rust to achieve those goals. We limit the number of I/O operations done in order to obtain relevant environment information and reduce the number of repeated I/O operations needed by gathering all relevant environment information immediately. This also reduces the need to spawn processes helping increase performance. 
```

# Python environment tools for Visual Studio Code

Performant Python environment tooling and support, such as locating all global Python installs and virtual environments.

## Environment Types Supported 

- python.org
- Windows Store
- PyEnv
- PyEnv-Win
- PyEnv-Virtualenv
- Conda
- Miniconda
- Miniforge
- PipEnv
- Homebrew
- VirtualEnvWrapper
- VirtualEnvWrapper-Win
- Venv
- VirtualEnv
- Python on your PATH

> - Mamba

Reviewer 2: Mamba now recomments using `miniforge or micromamba`, hence I'd not mention this.

Reviewer 2: Please remove this section

> - Micromamba

Reviewer 2: Lets leave this out for now, its not easy to get the exe for micromamba,
We can add this later.
We might end up discovering the envrionments, but perhaps not the conda exe.

Reviewer 2: Please remove this line

> - Mamba
> - Micromamba

Reviewer: ```suggestion
```
